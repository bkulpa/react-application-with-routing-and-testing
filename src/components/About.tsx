import { FC, ReactNode } from 'react';

export type TProps = {
    title: string;
    children: ReactNode;
};

export const About: FC<TProps> = ({ title, children }) => {
    return (
        <>
            <h2>{title}</h2>
            {children}
        </>
    )
}