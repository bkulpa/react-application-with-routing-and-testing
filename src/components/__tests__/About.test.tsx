import { render } from '@testing-library/react';
import { About } from '../About';

describe('About', () => {
    it('should render', () => {
        // before
        const { asFragment } = render(<About title="About Us">children</About>);
        // result
        expect(asFragment()).toMatchSnapshot();
    });
});