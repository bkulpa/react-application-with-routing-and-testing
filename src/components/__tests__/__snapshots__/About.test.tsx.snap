// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render 1`] = `
<DocumentFragment>
  <h2>
    About Us
  </h2>
  children
</DocumentFragment>
`;
